### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by anntzer**
I noticed while working on #9124 that some old codepaths seem to assume keys don't necessarily exist in rcParams, which may possibly have been the case a long time ago but is not the case anymore today -- so remove all rcParams.get by indexing into rcParams.

... except in axis3d which tries to look up the nonexistent ztick.major.width and ztick.color, so keep the fallback on xtick.major.width and xtick.color (I did not add the whole suite of ztick.* rcParams to rcsetup because in that case we'd better make sure all of them are respected, which is another project in itself).  Now that I was there I could as well do some style cleanups in axis3d.py.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_pgf.py [hunk 1]
```diff
@@ -223,13 +220,14 @@ def get_latex_manager():
         latex_header = LatexManager._build_latex_header()
         prev = LatexManagerFactory.previous_instance
 
-        # check if the previous instance of LatexManager can be reused
-        if prev and prev.latex_header == latex_header and prev.texcommand == texcommand:
-            if rcParams.get("pgf.debug", False):
+        # Check if the previous instance of LatexManager can be reused.
+        if (prev and prev.latex_header == latex_header
```
Comment (story645) [inline lib/matplotlib/backends/backend_pgf.py]: order of operations unclear here (and yes I know was unclear previously too), what about instead using parenthesis around each == pair?

Comment (anntzer) [reply lib/matplotlib/backends/backend_pgf.py]: I think it is pretty well known that "and" and "or" have lower priority than comparison operators.  Relying on this basically occurs everywhere throughout the codebase.

Comment (story645) [reply lib/matplotlib/backends/backend_pgf.py]: *shrugs* I never remember this stuff, but not gonna hold up the review over it.

Comment (tacaswell) [reply lib/matplotlib/backends/backend_pgf.py]: explicit is better than implicit, but too many `()` can also make it confusing by adding to visual noise.  In this case I think it is ok without `()`

**Pull request merged**